Design.md

```diff
@@ -172,7 +172,9 @@ This API creates a user on Vault.
 1. `create-user` api will be called using assumed role credentials.
     * The tenant user's `cdUserId` is stored as `username` in the corresponding Vault user.
     * The tenant user's `username` and `role` enum value are stored in the Vault user path as `<tenantUsername>::<roleEnumValue>`.
-1. `attach-policy` for the new user will be called using the assumed role credentials with full S3 and IAM access permissions.
+1. `get-policy` api will be called with policy name `userPolicy@[account-id]` using assumed role credentials
+1. If the `get-policy` API does not return any policy, using the assumed role credentials create an IAM managed policy with policy name `userPolicy@[account-id]` with full S3 and KMS access.
```

Comment:
Should this be full IAM access instead of KMS access?

Reply:
Users on OSIS does not need IAM access but as they should be able to create and manage keys.
These are the same permissions provided to the user by VMWARE OSE with AWS.

Reply:
sounds good, thanks for clarifying